review/module.py

```diff
@@ -0,0 +1,901 @@
+        review: TeacherReview,
+        sudo: bool = False,
+        delete: bool = False,
+    ) -> ReviewEmbed:
+        """Create teacher review embed"""
+        if review.anonym and not sudo:
+            name = _(ctx, "Anonymous")
+        else:
+            name = self.bot.get_user(review.author_id) or _(ctx, "Unknown user")
+
+        title = (
+            _(ctx, "Review for teacher #{id}:")
+            if not delete
+            else _(ctx, "Do you want to delete review for teacher #{id}?")
+        )
+
+        title = title.format(id=teacher.school_id)
+
+        embed = ReviewEmbed(
+            author=ctx.author,
+            title=title,
+            description=teacher.name if teacher.name else nextcord.Embed.Empty,
+            color=nextcord.Color.green(),
+            url=nextcord.Embed.Empty,
+            review=review,
+            bot=self.bot,
+        )
+
+        embed.add_field(
+            name=_(ctx, "Review ID #{id}").format(id=review.idx),
+            value=_(ctx, "Teacher grade: {grade}").format(grade=review.grade),
+            inline=False,
+        )
+
+        embed.add_field(
+            name=_(ctx, "Author:"),
+            value=name,
+        )
+
+        if review.created != review.updated:
+            embed.add_field(
+                name=_(ctx, "Updated:"),
+                value=review.updated,
+            )
+
+        embed.add_field(
+            name=_(ctx, "Date of review:"),
+            value=review.created,
+        )
+
+        text_review = textwrap.wrap(review.text_review, 1024)
+        name = _(ctx, "Text:")
+
+        for text in text_review:
+            embed.add_field(name=name, value=text, inline=False)
+            name = "\u200b"
+
+        embed.add_field(name="👍\u200b", value=review.upvotes)
+        embed.add_field(name="👎\u200b", value=review.downvotes)
+
+        return embed
+
+    @staticmethod
+    def _split_list(li, n) -> List[List]:
+        """Split list into lists of N items
+
+        Args:
+            li: List to split
+            n: Number of items in one list
+        """
+        for i in range(0, len(li), n):
+            yield li[i : i + n]
+
+    async def _add_subject_review(
+        self, ctx, abbreviation: str, grade: int, text: str, anonymous: bool
+    ) -> Optional[SubjectReview]:
+        """Add and return review.
+
+        Returns: review if success, None otherwise"""
+        subject = Subject.get_by_abbreviation(ctx, abbreviation)
+
+        if anonymous:
+            await utils.discord.delete_message(ctx.message)
+
+        if not subject:
+            await ctx.send(
+                _(ctx, "Subject with abbreviation {abbreviation} not found.").format(
+                    abbreviation=abbreviation
+                )
+            )
+            return
+
+        if grade < 1 or grade > 5:
+            await ctx.send(_(ctx, "Grade must be between 1 and 5 inclusive."))
+            return
+
+        if text is None or not len(text):
```

Comment:
Maybe consider adding a max length check? Same goes for Teacher reviews.

Reply:
I think this is not necessary since Discord limits message length and I'm counting with long review text when preparing review embed list.

Also, this will be reworked since DC plans to add some forms and inputs as I've seen.